src/cli/init/walkthrough.ts

```diff
@@ -4,19 +4,34 @@
 import { promises as fs } from "fs";
 import prompt, { Answers, PromptObject } from "prompts";
 
-import { Errors } from "../../lib/errors";
+import { ErrorList, Errors } from "../../lib/errors";
 import { eslint, gitignore, prettier, readme, tsConfig } from "./templates";
 import { sanitizeName } from "./utils";
 
-export type InitOptions = Answers<"name" | "description" | "errorBehavior">;
+export type PromptOptions = {
+  name: string;
+  description: string;
+  errorBehavior: "audit" | "ignore" | "reject";
+};
 
-export function walkthrough(): Promise<InitOptions> {
+export type PartialPromptOptions = Partial<PromptOptions>;
+
+export async function walkthrough(opts?: PartialPromptOptions): Promise<PromptOptions> {
+  const result = {
+    ...(await setName(opts?.name)),
+    ...(await setDescription(opts?.description)),
+    ...(await setErrorBehavior(opts?.errorBehavior)),
+  };
+  return result as PromptOptions; //TODO: Type coercion issue
```

Comment:
Fix TODO

Reply:
The specific error message is:
`Type '{ [x: string]: any; }' is missing the following properties from type 'PromptOptions': name, description, errorBehaviorts(2739)`

Reply:
We've chosen to accept this type coercion.